### ISSUE TITLE
[Feature] `/res update`, a command to expand a Residence without removing and recreating it.

### ISSUE THREAD

**Issue by clucky** [t00thpick1/Residence#17]
Currently, the only way to expand your residence is to delete it (currently not recieving any sort of refund) and recreate it. This can pose as a MAJOR inconvenience for a player who made their claim JUST ONE BLOCK too short. A great way to solve this inconvenience on players is to introduce a `/res update` command. There are two ways that this process can be done.

The first method would be used if you were wanting to expand or contract your residence. The player can start by typing `/res select existing [residence_name]`, which will set the residence selection (or [WorldEdit selection](https://github.com/t00thpick1/Residence/issues/18) if `ForceWorldEditSelection` is set to `true` in the configuration) to the positions of the residence specified in the last parameter (if none is specified, it will use the one that the player is standing in). The player can now use the selection commands to expand, contract or resize the selection. After the player has selected the desired area, they can finish the process by typing `/res update [residence_name]` (again, if they do not specify the last parameter, it uses the residence that they are standing in) which will prompt them to `/res confirm`.

The second would be used if you weren't expanding or contracting the residence, and instead, moving it completely. You can do this simply by selecting the new area that you want the residence to be "moved" to. The new selection does not have to be the same size as the residence beforehand, it will simply be resized to the new area. When the player is satisfied with the new area for their residence, they may type `/res update [residence_name]` to be prompted to `/res confirm` the update.

When a player is changing the size of their Residence, they have the choice to either make it larger or smaller. To avoid any cheating, the player will have money deducted or refunded to their account after updating their residence (they will be notified how much will be deducted/refunded in the `/res confirm` message. When a player is being refunded money, the plugin will calculate in the refund percentage ([suggested by WoodenAxe](https://github.com/t00thpick1/Residence/issues/7)), so not to bypass the `/res remove` command. As well, if any subzones are outside of the boundaries of the parent residence after the update, these will be removed upon confirmation (the player will be warned of this in the `/res confirm` message).

I believe `/res update` is a feature that should have been in Residence from the very beginning, being follows the best interests of the player.

---

Check comments for additional information.

**Comment by clucky** [comment]
Method for `/res update` edited on 8/6/2014. Please read changes.

**Comment by hector-romero** [comment]
Hi.
You can actually expand your res by making the updated selection (using any method) and then doing: `/res area replace <resname> main` . It will replace the main area (wich in most cases is the only area in a res) by the new selection, and you don't have to pay again for the blocks you already own.

Regards!

**Comment by clucky** [comment]
Yes, but that feature only exists in version 2, which is now going obsolete.

**Comment by clucky** [comment]
As well, be sure to look out for this potential bug:

When a user is attempting to expand a subzone within a residence that they do not have admin permissions to, deny the update. Neglecting to block this bug could result in players purchasing smaller subzones within a city, and expand it to cover a larger area.

**Comment by clucky** [comment]
As well, also make sure to prevent users from expanding subzones into other subzones or expanding subzones outside of the residence.

**Comment by clucky** [comment]
If any subzones are outside of the new area after the update, the player will be prompted for them to be removed in order to proceed with the update. This goes for all subzones except those which are either not owned by the player, or the player does not have the `admin` flag for.

`/res update` will comply with issue #19
